pkg/trace/api/otlp.go

```diff
@@ -451,49 +465,12 @@ func spanKind2Type(kind otlppb.Span_SpanKind, span *pb.Span) string {
 	return typ
 }
 
-func byteArrayToUint64(b []byte) uint64 {
-	if len(b) < 8 {
-		return 0
-	}
+func traceIDToUint64(b [16]byte) uint64 {
 	return binary.BigEndian.Uint64(b[len(b)-8:])
 }
 
-// anyValueString converts otlppb.AnyValue a to its string representation.
-func anyValueString(a *otlppb.AnyValue) string {
```

Comment:
There are very small but negligible differences for `Array` and `Kvlist` value. For example, we previously had `a:1,b:2` and now it would be `{"a": 1, "b": 2}`. I think the latter is much better as JSON and matches what we do with events. I'd keep it as is. WDYT?

Reply:
I think it's better to use `AsString` if we can, but my question was more "Could that break an end-user?"

Reply:
Well, yes. Technically it could. We don't know how many people use this currently, and how many of those rely on the format of "array" and "kv" attribute values.

Reply:
Okay! I think this could as well be considered an 'enhancement' so I would propose we list this on the changelog as that with a note like 'OTLP attributes stringification is now consistent with the OpenTelemetry Collector one' (not sure if 'stringification' is the right word here but I don't have better ideas 😄)

Reply:
Done! PTAL if you think the phrasing is OK